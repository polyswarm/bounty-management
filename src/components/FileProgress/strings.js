const strings = {
  remove: 'Remove'
};

export default strings;
