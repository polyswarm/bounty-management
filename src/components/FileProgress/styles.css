.FileProgress {
  background-color: #eee;
  height: 72px;
  width: 100%;
}
