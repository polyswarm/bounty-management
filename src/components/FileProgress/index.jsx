import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Component Imports
import strings from './strings';
import './styles.css';

class FileProgress extends Component {
  constructor(props) {
    super(props);
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  render () {
    const { props: { file, progress} } = this;
    return(
      <span className='FileProgress'>
        <div>
          <p className='FileProgress-Name'>{file.name}</p>
          <progress>{progress}</progress>
        </div>
        <button onClick={this.onClickHandler}>
          {strings.remove}
        </button>
      </span>
    );
  }

  onClickHandler() {
    const { props: { remove } } = this;
    if (remove) {
      remove();
    }
  }
}

FileProgress.proptypes = {
  file: PropTypes.string.isRequired,
  progress: PropTypes.number,
  remove: PropTypes.func.isRequired,
};

export default FileProgress;
