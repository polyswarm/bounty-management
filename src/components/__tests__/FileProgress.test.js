import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import FileProgress from '../FileProgress';

it('throws on render without files', () => {
  expect(() => {
    render(<FileProgress />);
  }).toThrow();
});

it('renders without crashing', () => {
  const file = {
    name: 'asdf'
  };
  const wrapper = render(<FileProgress file={file} />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

if('calls the mock function when I click close', () => {
  const remove = jest.fn();
  const file = {
    name: 'asdf'
  };
  const wrapper = render(<FileProgress file={file} remove={remove} />);
  wrapper.find('button').simulate('click');
  expect(remove).toHaveBeenCalled();
});
