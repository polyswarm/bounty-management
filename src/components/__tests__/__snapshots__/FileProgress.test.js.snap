// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span
  class="FileProgress"
>
  <div>
    <p
      class="FileProgress-Name"
    >
      asdf
    </p>
    <progress>
      
    </progress>
  </div>
  <button>
    Remove
  </button>
</span>
`;
